src/mono/wasm/Wasm.Build.Tests/Common/ToolCommand.cs

```diff
@@ -109,24 +117,30 @@ private async Task<CommandResult> ExecuteAsyncInternal(string executable, string
             CurrentProcess = CreateProcess(executable, args);
             DataReceivedEventHandler errorHandler = (s, e) =>
             {
-                if (e.Data == null)
-                    return;
-
-                string msg = $"[{_label}] {e.Data}";
-                output.Add(msg);
-                _testOutput.WriteLine(msg);
-                ErrorDataReceived?.Invoke(s, e);
+                lock (_lock)
+                {
+                    if (e.Data == null || isDisposed)
+                        return;
```

Comment:
nit: Typically, the pattern is like if-lock-if, but I guess it shouldn't that important here

```suggestion
                if (e.Data == null || isDisposed)
                    return;

                lock (_lock)
                {
                    if (e.Data == null || isDisposed)
                        return;
```